content/influxdb/v2.0/reference/internals/shards.md

```diff
@@ -0,0 +1,159 @@
+A single shard contains multiple series, multiple [TSM files](#tsm-time-structured-merge-tree) on disk,
+and belongs to a [shard group](#shard-groups).
+
+## Shard groups
+Shard groups are logical containers for InfluxDB shards.
+A single shard group contains time series data for a specific time range defined by
+the [shard group duration](#shard-group-duration).
+A shard group belongs to an InfluxDB [bucket](/influxdb/v2.0/reference/glossary/#bucket).
+
+{{% note %}}
+In **InfluxDB OSS**, a shard group typically contains only a single shard.
+In an **InfluxDB Enterprise 1.x cluster**, shard groups contain multiple shards
+distributed across multiple data nodes.
+{{% /note %}}
+
+### Shard group duration
+The **shard group duration** determines the time range a shard group covers
+and the interval at which new shard groups are created.
+By default, InfluxDB determines shard group durations using
+the [retention period](/influxdb/v2.0/reference/glossary/#retention-period)
+of the bucket:
+
+| Bucket retention period     | Default shard group duration |
+|:-----------------------     | ----------------------------:|
+| less than 2 days            | 1h                           |
+| between 2 days and 6 months | 1d                           |
+| greater than 6 months       | 7d                           |
+
+#### Manage shard group durations
+To configure a custom bucket shard group duration, use the `--shard-group-duration`
+flag with the [`influx bucket create`](/influxdb/v2.0/reference/cli/influx/bucket/create/#create-a-custom-shard-group-duration)
+and [`influx bucket update`](/influxdb/v2.0/reference/cli/influx/bucket/update//#update-the-shard-group-duration-of-a-bucket) commands.
+
+{{% note %}}
+Shard group durations must be shorter than the bucket's retention period.
+{{% /note %}}
+
+To view your bucket's shard group duration, use the
+[`influx bucket list` command](/influxdb/v2.0/reference/cli/influx/bucket/list/).
+
+### Shard group diagram
+The following diagram represents a **bucket** with a **4d retention period**
+and a **1d shard group duration**:
+
+---
+
+{{< html-diagram/shards >}}
+
+---
+
+## Shard life-cycle
+
+### Shard precreation
+The InfluxDB **shard precreation service** creates new shards before data actually arrives.
+The precreator service only creates shards with future start and end times.
+It does not create shards that are either wholly or partially in the past.
+
+#### Configure shard precreation
+To configure the InfluxDB shard precreation service, use the following `influxd`
+configuration settings:
+
+- [`storage-shard-precreator-advance-period`](/influxdb/v2.0/reference/config-options/#storage-shard-precreator-advance-period)
+- [`storage-shard-precreator-check-interval`](/influxdb/v2.0/reference/config-options/#storage-shard-precreator-check-interval)
+
+### Shard states
+A shard can be in one of two states: **hot** or **cold**.
```

Comment:
Hmmm, I'm not sure about this. A user can write into any shard they like regardless of whether it's the most recent shard in the shard-group or not. I'm not sure what value this section will have for an end-user. What are we trying to convey? Is it that you should avoid writing into into old shards that have been fully compacted?

Reply:
No, just that these states exists. We use this terminology a lot, but we don't explain it anywhere. What if I remove the compaction phrases?

Reply:
Do we need to use the term hot/cold, or just convey the shard is being written to or not? Simplifies verbiage and mental model if we boil it down to that. Then we could remove this section, and include the following text in next section:

_Typically, InfluxDB writes data to the most recent shard group, but when backfilling
historical data, older shards <may be/are?> written to. When the backfill is complete, InfluxDB re-compacts the older shards._

Reply:
In the 2.0 docs, we do not because this is the first real attempt to explain shards in the 2.0 docs. We do when talking about shards in 1.x. The terms "hot" and "cold" are common when talking about InfluxDB shards.

Reply:
Thanks, @sanderson. I'd lean toward omitting terms like this when possible. If hot/cold conveyed multiple criteria, that'd make sense. Otherwise, it's extra term for folks to look up.

Reply:
The original reason for including these terms was that these terms are commonly used when talking about shard management with customers. Our support and product teams both use them to describe the state of shards and these states have other implications.

For example, recently an Enterprise customer was seeing performance dips in their database and it appeared to be a disk IO issue. They contacted their underlying host provider who analyzed the actual physical disks and couldn't find any issues. What it came down to is that they were backfilling data and didn't understand that InfluxDB had to un-compact the old (cold) shards to write to them (making them hot), then re-compact them, which can be a very disk IO-heavy process.

I hesitate to omit these terms because they're used so commonly by our support and product teams.

Reply:
I understand it's familiar jargon, but think it's just as clear w/o:
_InfluxDB must un-compact old shards to write to them, and then re-compact the shard afterwards, which can be a very disk IO-heavy process._

Typically, best to omit jargon.